Create CONTRIBUTION_LADDER.md (#791)
cncf/cnf-testsuite: CONTRIBUTION_LADDER.md

# Contribution Ladder

---
* [Community Member](#community-member)
* [Contributor](#contributor)

>   * [How to become a contributor](#how-to-become-a-contributor)

Author: @agentpoyo suggest edit please :)

* [Maintainer](#maintainer)
  * [How to become a maintainer](#how-to-become-a-maintainer)
* [Admin](#admin)
  * [How to become an admin](#admin)
---

Our ladder defines the roles and responsibilities on this project and how to
participate with the goal of moving from a user to a maintainer. You will need
to gain people's trust, demonstrate your competence and understanding, and meet
the requirements of the role.

## Community Member

Everyone is a community member! 😄 You've read this far so you are already ahead. 💯

Here are some ideas for how you can be more involved and participate in the community:

* Comment on an issue that you’re interested in.
* Submit a pull request to fix an issue.
* Report a bug.

> * Share a bundle that you made and how it went.

Reviewer: ```suggestion
```

* Come chat with us in [Slack][https://cloud-native.slack.com/archives/C014TNCEX8R].

> You must follow our [Code of Conduct](code-of-conduct.md).

Reviewer 2: Is there a better way to say "You must..."? Maybe something like "It is fundamental to follow our [Code of Conduct](code-of-conduct.md) as a contributor".

> You must follow our [Code of Conduct](code-of-conduct.md).

Author: ```suggestion
Community members must follow our [Code of Conduct](code-of-conduct.md).
```

How about this @agentpoyo

Reviewer 2: That does sound better. The word "must" just seems very authoritarian to me.

## Contributor

[Contributors][contributors] have the following capabilities:

* Have issues and pull requests assigned to you
* Apply labels, milestones and projects
* [Mark issues as duplicates](https://help.github.com/en/articles/about-duplicate-issues-and-pull-requests)
* Close and assign issues and pull requests

> Contributors must agree to and follow this Contributing Guide.

Reviewer 2: How about:  "We ask contributors to agree and follow this Contributing Guide" @taylor

> Contributors must agree to and follow this Contributing Guide.

Author: @agentpoyo sounds good. Add a suggest edit please

> Contributors must agree to and follow this Contributing Guide.

Author: new wording just says follow ...

> Contributors must agree to and follow this Contributing Guide.

Reviewer 2: ```suggestion
We ask that contributors agree to and follow this Contributing Guide.
```

### How to become a contributor

To become a contributor, the maintainers of the project would like to see you:

* Comment on issues with your experiences and opinions.
* Add your comments and reviews on pull requests.
* Contribute pull requests.
* Open issues with bugs, enhancement ideas, and questions.

Contributors and maintainers will do their best to watch for community members

> who may make good contributors. But don’t be shy, if you feel that this is you,

Reviewer 2: ```suggestion
who make good contributors. But don’t be shy, if you feel that this is you,
```

please reach out to one or more of the contributors or [maintainers](MAINTAINERS.md).

## Maintainer

[Maintainers][maintainers] are members with extra capabilities:

* Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested.
* Review pull requests.
* Merge pull requests.

Maintainers also have additional responsibilities beyond just merging code:

* Help foster a safe and welcoming environment for all project participants.
  This will include understanding and enforcing our [Code of Conduct](code-of-conduct.md).
* Organize and promote pull request reviews, e.g. prompting community members,
  contributors, and other maintainers to review.
* Triage issues, e.g. adding labels, promoting discussions, finalizing decisions.
* Help organize our development meetings, e.g. schedule, organize and
  execute agenda.

> Maintainers must agree to and follow the [Reviewing Guide](REVIEWING.md).

Author: @agentpoyo add a suggest edit please similar to your contributor suggestion above https://github.com/cncf/cnf-testsuite/pull/791#discussion_r641698992

> Maintainers must agree to and follow the [Reviewing Guide](REVIEWING.md).

Author: new wording just says follow ...

# Contribution Ladder

---
* [Community Member](#community-member)
* [Contributor](#contributor)
  * [How to become a contributor](#how-to-become-a-contributor)
* [Maintainer](#maintainer)
  * [How to become a maintainer](#how-to-become-a-maintainer)
* [Admin](#admin)
  * [How to become an admin](#admin)
---

Our ladder defines the roles and responsibilities on this project and how to
participate with the goal of moving from a user to a maintainer. You will need
to gain people's trust, demonstrate your competence and understanding, and meet
the requirements of the role.

## Community Member

Everyone is a community member! 😄 You've read this far so you are already ahead. 💯

Here are some ideas for how you can be more involved and participate in the community:

* Comment on an issue that you’re interested in.
* Submit a pull request to fix an issue.
* Report a bug.

> * Share a bundle that you made and how it went.

Author: No clue. Let's delete it.

They must follow our [Code of Conduct](CODE_OF_CONDUCT.md).

[slack]: https://slack.cncf.io/ #cnf-testsuite-dev

## Contributor

[Contributors][contributors] have the following capabilities:

* Have issues and pull requests assigned to them
* Apply labels, milestones and projects
* [Mark issues as duplicates](https://help.github.com/en/articles/about-duplicate-issues-and-pull-requests)
* Close, reopen, and assign issues and pull requests

They must agree to and follow this Contributing Guide.

### How to become a contributor

To become a contributor, the maintainers of the project would like to see you:

* Comment on issues with your experiences and opinions.
* Add your comments and reviews on pull requests.
* Contribute pull requests.
* Open issues with bugs, experience reports, and questions.

Contributors and maintainers will do their best to watch for community members
who may make good contributors. But don’t be shy, if you feel that this is you,
please reach out to one or more of the contributors or maintainers.

[contributors]: TBD <!-- https://github.com/orgs/getporter/teams/contributors -->

## Maintainer

[Maintainers][maintainers] are members with extra capabilities:

> * Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested.

Author: Added .github/CODEOWNERS to this PR. This makes a PR request automatically add those people as reviewers.  See https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners

> * Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested.

Author: @lixuna 👆🏻

> * Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested.

Reviewer: ```suggestion
* Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested
```

* Review pull requests.
* Merge pull requests.

Maintainers also have additional responsibilities beyond just merging code:

* Help foster a safe and welcoming environment for all project participants.
  This will include understanding and enforcing our [Code of Conduct](CODE_OF_CONDUCT.md).
* Organize and promote pull request reviews, e.g. prompting community members,
  contributors, and other maintainers to review.
* Triage issues, e.g. adding labels, promoting discussions, finalizing decisions.
* Help organize our development meetings, e.g. schedule, organize and
  execute agenda.

They must agree to and follow the [Reviewing Guide](REVIEWING.md).

[maintainers]: TBD <!-- https://github.com/orgs/getporter/teams/maintainers -->

### How to become a maintainer

To become a maintainer, we would like you to see you be an effective
contributor, and show that you can do some of the things maintainers do.
Maintainers will do their best to regularly discuss promoting contributors. But
don’t be shy, if you feel that this is you, please reach out to one or more of
the maintainers.

## Admin

[Admins][admins] are maintainers with extra responsibilities:

> TBD

Reviewer: ```suggestion
```

> * Create new mixin repositories

Reviewer: ```suggestion
```

* [Maintainer](#maintainer)
  * [How to become a maintainer](#how-to-become-a-maintainer)
* [Admin](#admin)
  * [How to become an admin](#admin)
---

Hello! We are excited that you want to learn how to get involved with the CNF Test Suite! This contributor ladder outlines the different contributor roles within the project, along with the responsibilities and privileges that come with them. Community members generally start at the first levels of the "ladder" and advance up it as their involvement in the project grows. Our project members are happy to help you advance along the contributor ladder.

Each of the contributor roles below is organized into lists of three types of things. "Responsibilities" are things that contributor is expected to do. "Requirements" are qualifications a person needs to meet to be in that role, and "Privileges" are things contributors on that level are entitled to.

You will need to gain people's trust, demonstrate your competence and understanding, and meet the requirements of the role.

## Community Member

Everyone is a community member! 😄 You've read this far so you are already ahead. 💯

How you can learn more?

> * Read the [install guide](INSTALL.md), [usage guide](USAGE.md), [FAQ](FAQ.md), and other documentation.

Reviewer: ```suggestion
* Read the [install guide](INSTALL.md), [usage guide](USAGE.md), [FAQ](FAQ.md), and other documentation
```

> * Request a live demonstration and introduction of the test suite

Reviewer: ```suggestion
* Request a live demonstration and introduction of the test suite ([book a time on Calendly](https://calendly.com/cnftestsuite))
```

> * Come chat with us in [Slack](https://cloud-native.slack.com/archives/C014TNCEX8R).

Author: Add a suggest edit @agentpoyo

> * Come chat with us in [Slack](https://cloud-native.slack.com/archives/C014TNCEX8R).

Reviewer 2: ```suggestion
* Come chat with us in [Slack](https://cloud-native.slack.com/archives/C014TNCEX8R) or start a [discussion](https://github.com/cncf/cnf-testsuite/discussions).
```

> * Come chat with us in [Slack](https://cloud-native.slack.com/archives/C014TNCEX8R) or start a [discussion](https://github.com/cncf/cnf-testsuite/discussions).

Reviewer: ```suggestion
* Come chat with us in [Slack](https://cloud-native.slack.com/archives/C014TNCEX8R) or start a [discussion](https://github.com/cncf/cnf-testsuite/discussions)
```

## Community Participant

A Community Participant engages with the project and its community, contributing their time, thoughts, etc. Community participants are usually users who have stopped being anonymous and started being active in project discussions.

Responsibilities:
- Follow our [Code of Conduct](code-of-conduct.md)

How you can be more involved and participate in the community:

* Participating in community discussions
* Helping other users
* Commenting on issues
* Trying out new releases
* Attending [community meetings](https://github.com/cncf/cnf-testsuite#communication-and-community-meetings) and events

## Contributor

[Contributors][https://github.com/cncf/cnf-testsuite/blob/main/MAINTAINERS.md#contributors] have the following capabilities:

* Have issues and pull requests assigned to you
* Apply labels, milestones and projects
* [Mark issues as duplicates](https://help.github.com/en/articles/about-duplicate-issues-and-pull-requests)
* Close and assign issues and pull requests

> Contributors will agree to and follow this Contributing Guide.

Reviewer: ```suggestion
Contributors will agree to and follow this [Contributing Guide](CONTRIBUTING.md).
```

### How to become a contributor

To become a contributor, the maintainers of the project would like to see you:

> * Comment on issues with your experiences and opinions.

Reviewer: ```suggestion
* Comment on issues with your experiences and opinions
```

> * Add your comments and reviews on pull requests.

Reviewer: ```suggestion
* Add your comments and reviews on pull requests
```

> * Contribute pull requests.

Reviewer: ```suggestion
* Contribute pull requests
```

> * Open issues with bugs, enhancement ideas, and questions.

Reviewer: ```suggestion
* Open issues with bugs, enhancement ideas, and questions
```

Contributors and maintainers will do their best to watch for community members
who make good contributors. But don’t be shy, if you feel that this is you,
please reach out to one or more of the contributors or [maintainers](MAINTAINERS.md).

## Maintainer

[Maintainers][https://github.com/cncf/cnf-testsuite/blob/main/MAINTAINERS.md#maintainers] are members with extra capabilities:

* Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested.

> * Review pull requests.

Reviewer: ```suggestion
* Review pull requests
```

> * Merge pull requests.

Reviewer: ```suggestion
* Merge pull requests
```

Maintainers also have additional responsibilities beyond just merging code:

> * Help foster a safe and welcoming environment for all project participants.

Reviewer: ```suggestion
* Help foster a safe and welcoming environment for all project participants
```

>   This will include understanding and enforcing our [Code of Conduct](code-of-conduct.md).

Reviewer: ```suggestion
  This will include understanding and enforcing our [Code of Conduct](code-of-conduct.md)
```

* Organize and promote pull request reviews, e.g. prompting community members,

>   contributors, and other maintainers to review.

Reviewer: ```suggestion
  contributors, and other maintainers to review
```

> * Triage issues, e.g. adding labels, promoting discussions, finalizing decisions.

Reviewer: ```suggestion
* Triage issues, e.g. adding labels, promoting discussions, finalizing decisions
```

* Help organize our development meetings, e.g. schedule, organize and

>   execute agenda.

Reviewer: ```suggestion
  execute agenda
```

> We ask that Maintainers agree to and follow the [Reviewing Guide](REVIEWING.md).

Reviewer: delete: we don't have a reviewing guide yet

### How to become a maintainer

To become a maintainer, we would like to see you be an effective
contributor, and show that you can do some of the things maintainers do.
Maintainers will do their best to regularly discuss promoting contributors. But
don’t be shy, if you feel that this is you, please reach out to one or more of
the maintainers.

## Admin

[Admins][MAINTAINERS.md#admin] are maintainers with extra responsibilities:

* Manage repositories
* Manage CODEOWNERS file
* Manage GitHub Actions infra (eg. runners)
* Manage teams

### How to become an admin

It isn't expected that all maintainers will need or want to move up to admin. If
you are a maintainer, and find yourself often asking an admin to do certain
tasks for you and you would like to help out with administrative tasks, please

> reach out to one or more of the admins.

Reviewer: ```suggestion
reach out to one or more of the admins.

## Emeritus Maintainers
Former maintainers are recognized with an honorary Emeritus Maintainer status, and have their names permanently listed in the [maintainer file](MAINTAINERS.md) as a form of gratitude for their contributions.
```

* Add your comments and reviews on pull requests
* Contribute pull requests
* Open issues with bugs, enhancement ideas, and questions

Contributors and maintainers will do their best to watch for community members
who make good contributors. But don’t be shy, if you feel that this is you,
please reach out to one or more of the contributors or [maintainers](MAINTAINERS.md).

## Maintainer

[Maintainers][https://github.com/cncf/cnf-testsuite/blob/main/MAINTAINERS.md#maintainers] are members with extra capabilities:

* Be a [Code Owner](.github/CODEOWNERS) and have reviews automatically requested
* Review pull requests
* Merge pull requests

Maintainers also have additional responsibilities beyond just merging code:

* Help foster a safe and welcoming environment for all project participants
  This will include understanding and enforcing our [Code of Conduct](code-of-conduct.md)
* Organize and promote pull request reviews, e.g. prompting community members,
  contributors, and other maintainers to review
* Triage issues, e.g. adding labels, promoting discussions, finalizing decisions
* Help organize our development meetings, e.g. schedule, organize and
  execute agenda

> We ask that Maintainers agree to and follow the [Reviewing Guide](REVIEWING.md).

Author: ```suggestion
```

deleting at @lixuna suggestion
